File: leetcode beginners/reversing string prefix.py
class Solution:
    def reversePrefix(self, s: str, k: int) -> str:
        
        stack = []
        result = []
        for i in range(0, k):
            stack.append(s[i])

        while stack:
            result.append(stack.pop())

        if k == len(s):
            return ''.join(result)
        else:
            for i in range(k,len(s)):
                result.append(s[i])

        return ''.join(result)

        
        

File: leetcode in c/reversing string prefix.c
char* reversePrefix(char* s, int k) {
    
    for (int i = 0; i < k / 2; i++)
    {
        int tmp = s[i];
        s[i] = s[k-1-i];
        s[k-1-i] = tmp;
    }

    return s;
}